src/native/eventpipe/ep-provider.c

```diff
@@ -76,7 +76,7 @@ provider_prepare_callback_data (
 
 	return ep_provider_callback_data_init (
 		provider_callback_data,
-		filter_data,
+		ep_rt_utf8_string_dup (filter_data),
```

Comment:
Wouldn't it be better to duplicate this string as part of ep_provider_callback_data_init since fini will free this string, so ownership is held explicitly inside the ep_provider_callback_data and shouldn't be split between caller init call and internal implementation of fini.